registry/config_test.go

```diff
@@ -317,3 +319,57 @@ func TestNewServiceConfig(t *testing.T) {
 		}
 	}
 }
+
+func TestValidateIndexName(t *testing.T) {
+	ErrorString := "invalid index name (%s). Cannot begin or end with a hyphen"
+	testCases := []struct {
+		index  string
+		expect string
+		err    error
+	}{
+		{
+			index:  "index.docker.io",
+			expect: "docker.io",
+		},
+		{
+			index:  "example.com",
+			expect: "example.com",
+		},
+		{
+			index:  "127.0.0.1",
+			expect: "127.0.0.1",
+		},
+		{
+			index: "index.docker.io-",
+			err:   fmt.Errorf(ErrorString, "index.docker.io-"),
+		},
+		{
+			index: "-example.com",
+			err:   fmt.Errorf(ErrorString, "-example.com"),
+		},
+		{
+			index: "-127.0.0.1-",
+			err:   fmt.Errorf(ErrorString, "-127.0.0.1-"),
+		},
+		{
+			index:  "mytest-1.com",
+			expect: "mytest-1.com",
+		},
+	}
+
+	for _, testCase := range testCases {
+		s, err := ValidateIndexName(testCase.index)
+		if err != nil {
```

Comment:
This is a little bit "too complicated" (although it's not that complicated).
I think we could split this test into 2 (cases that errors out, and cases that passes).